src/dl.cpp

```diff
@@ -82,20 +102,43 @@ dl_error_t DL_DLL_EXPORT dl_instance_load_inplace( dl_ctx_t       dl_ctx,
 {
 	dl_data_header* header = (dl_data_header*)packed_instance;
 
-	if( packed_instance_size < sizeof(dl_data_header) ) return DL_ERROR_MALFORMED_DATA;
-	if( header->id == DL_INSTANCE_ID_SWAPED )           return DL_ERROR_ENDIAN_MISMATCH;
-	if( header->id != DL_INSTANCE_ID )                  return DL_ERROR_MALFORMED_DATA;
-	if( header->version != DL_INSTANCE_VERSION )        return DL_ERROR_VERSION_MISMATCH;
-	if( header->root_instance_type != type_id )         return DL_ERROR_TYPE_MISMATCH;
+	if( packed_instance_size < sizeof(dl_data_header) )                  return DL_ERROR_MALFORMED_DATA;
+	if( header->id == DL_INSTANCE_ID_SWAPED )                            return DL_ERROR_ENDIAN_MISMATCH;
+	if( header->id != DL_INSTANCE_ID )                                   return DL_ERROR_MALFORMED_DATA;
+	if( header->version != DL_INSTANCE_VERSION && header->version != 1 ) return DL_ERROR_VERSION_MISMATCH;
+	if( header->root_instance_type != type_id )                          return DL_ERROR_TYPE_MISMATCH;
 
-	const dl_type_desc* type = dl_internal_find_type(dl_ctx, header->root_instance_type);
-	if( type == 0x0 )
-		return DL_ERROR_TYPE_NOT_FOUND;
+	if( header->version == DL_INSTANCE_VERSION && !header->need_slow_patching )
+	{
+		uintptr_t offset_shift     = sizeof( uintptr_t ) * 4;
+		uintptr_t offset_mask      = ( 1ULL << offset_shift ) - 1;
+		uint8_t* patch_mem         = (uint8_t*)packed_instance;
+		uintptr_t offset_to_pointer = header->first_pointer_to_patch; // 0 is the patch terminator
+		while( offset_to_pointer != 0 )
+		{
+			patch_mem += offset_to_pointer;
```

Comment:
same issue here, extract it to a function? could probably be a function that just returns an error-code + have an error-code for DL_NEED_FULL_PATCH or similar and embed the version check and flag-check to get all of that in the same function and in one place.